src/generators/Sequence.js

```diff
@@ -4,27 +4,30 @@
 
 import Generator from './Generator';
 
-class Sequence extends Generator {
+export default class Sequence extends Generator {
   static sequences = {};
-  id = '';
 
-  constructor(factoryGirl, id, callback = null) {
+  constructor(factoryGirl, id = null, callback = null) {
     super(factoryGirl);
-
-    if (typeof id !== 'string') {
-      throw new Error('Invalid sequence key passed');
+    if (typeof id === 'function') {
+      callback = id;
+      id = null;
     }
-
-    this.id = id;
-
-    Sequence.sequences[id] = Sequence.sequences[id] || 1;
+    this.id = id || generateId();
     this.callback = callback;
+    Sequence.sequences[this.id] = Sequence.sequences[this.id] || 1;
   }
-
   generate() {
-    const count = Sequence.sequences[this.id]++;
-    return Promise.resolve(this.callback ? this.callback(count) : count);
+    const next = Sequence.sequences[this.id]++;
+    return this.callback ? this.callback(next) : next;
   }
 }
 
-export default Sequence;
+function generateId() {
+  let id;
+  let i = 0;
+  do {
+    id = `_${i++}`;
+  } while (id in Sequence.sequences);
+  return id;
+}
```

Comment:
@aexmachina, this fixes the problem, but this still needs some consideration:

We don't really need the generateId() method: There would be just one instance of Sequence generator (created in the generatorThunk), and only the first call to generate() without an id will result in the generateId() call. The side effect of this is also that all sequences without an id will share same sequence.

I guess the primary reason of having `id` optional was to be back compatible, the problem now is that it would change the generated sequences and any test relying on the generated sequence will now end up failing. Also, any new test may end up breaking existing tests since it may now change sequence generated for existing tests.

Given above points and that id will need to be mandatory for function initialisers, I guess it would be less confusing to make id mandatory for sequences. What do you think?